File: 0242-valid-anagram/0242-valid-anagram.cpp
#include <map>
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.length() != t.length()){
            return false;
        }
        std::map<char, int> wordCount2;
        std::map<char, int> wordCount1;
        for(int i = 0; i < s.length(); i++){
            wordCount2[s[i]] += 1;
            wordCount1[t[i]] += 1;

        }
        return wordCount2 == wordCount1;
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        def add_c(c, base_map):
            if c in base_map:
                base_map[c] += 1
            else:
                base_map[c] = 1
        if len(s) != len(t):
            return False
        s_map = {}
        t_map = {}
        for (s_c, t_c) in zip(s,t):
            add_c(s_c, s_map)
            add_c(t_c, t_map)
        if s_map == t_map:
            return True
        else:
            return False


        